### ISSUE TITLE
Documentation for Kore service and its API

### ISSUE THREAD

**Issue by DeepCowProductions** [jeflem/ananke#31]
Hi, while browsing the web for nbgrader improvements and integrations for moodle, I found your Project and was wondering whether there is any documentation on the Kore service that you use as a bridge between nbgrader and LTI 1.3.

 I thought about implementing this particular part myself, but that might not be necessary when is work has already been done.

In particular, I am wondering how nicely it could integrate with nbgraders backend alternatives for formgrader such as [ngshare](https://ngshare.readthedocs.io/en/latest/).

**Comment by mw07c6** [comment]
Hi @DeepCowProductions,

Thank you for bringing this up!

You're right—there isn't any official documentation for the Kore service at the moment. I'm currently working on it, and I hope to have something available soon.

As for your question about using Kore with ngshare, I think it may be possible with some tinkering. However, I don't have any experience with ngshare myself, so I can't say for sure how well they'd work together. If you do decide to give it a try, I'd love to hear about your results!

Please feel free to reach out if you have any other questions.

**Comment by DeepCowProductions** [comment]
Am I correct to assume that your [kore-extension]( https://github.com/marcwit/kore-extension) is the "frontend" for said service under [assets/kore](https://github.com/jeflem/ananke/tree/main/images/ananke-nbgrader/assets/kore)?

I am not quite sure how I would even install all required packages right now and configure them to work with our hub and moodle platform.  Just digging through the code right now, it seems at least possible.

In the meantime, may I suggest splitting the python kore-service off into a separate repository (setup.py or pyproject) so that one can install it via pip?

**Comment by jeflem** [comment]
> Am I correct to assume that your kore-extension is the "frontend" for said service under assets/kore?

Yes, you are. But the backend ('Kore service') is tightly integrated into JHub's config files. Some of the code runs as a separate (hub managed) service. But there's a lot of code which has to be placed in Jhub's config files (using the post auth hook, for instance). Untangling this close relationship is on the long-term todo list (maybe next year), but requires a well thought out concept and a collection of use cases to cover. Maybe we'll first start a discussion with the JHub community on this.

At the moment I do not see how to split the Kore service off into a separate repository. Acutally we had planned to develop such a separate service which brings LTI (bidirectional) to JHub and nbgrader. But it turned out that the LTI stuff requires too many modifications on JHub and nbgrader sides. Thus, we were forced to provide a whole Jupyter distribution. Without that, installing Kore would be a mess.
